### PR TITLE
Les 286 login page integration

### PR DESCRIPTION
**Pull request by Kurilo-Pavel**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -2,26 +2,47 @@ import "./App.css";
...
+        <div className="App">
+          <Body onLanguageSwitch={setLanguageCode}/>
+          <Routes>
+            <Route path='/users/sign_in' element={<LoginPage/>}/>
+            <Route path='/users/sign_in/phone_number' element={<PhoneNumberPage/>}/>
+            <Route path='/users/sign_in/phone_number/code' element={<CodePage/>}/>
+            <Route path='/search' element={<Search/>}/>
+          </Routes>
+
+          <div style={{display: 'flex', flexDirection: 'column'}}>
```
Comment (OxygenO22) [inline src/App.tsx:36]: It is better don't use inline styles

Comment (Kurilo-Pavel) [reply src/App.tsx:36]: this button for testing

File: src/components/PasswordAndConfirm.tsx [hunk 2]
```diff
@@ -29,14 +29,14 @@ const PasswordAndConfirm = ({ minSymbol, maxSymbol, isConfirm, field, error }: P
...
-      <label className='passwordLabel'>{isConfirm ? 'Confirm password' : 'Password'}
+      <label className='password-label'>{isConfirm ? 'Confirm password' : 'Password'}
         <input type={visiblePassword ? 'text' : 'password'}
-          className={`passwordInput ${error ? `errorInput` : ``}`}
-          minLength={minSymbol}
-          maxLength={maxSymbol}
-          {...field}
-          required />
-        <img className='image' alt='eye' src={visiblePassword ? open_eye : close_eye} onClick={showPassword} />
+               className={`password-input ${error ? `error-input` : ``}`}
```
Comment (Terapool) [inline src/components/PasswordAndConfirm.tsx:34]: use classnames

File: src/components/body/footer/Footer.tsx [hunk 3]
```diff
@@ -1,4 +1,55 @@
...
+  return (
+    <div style={{display:'flex'}}>
+      <Link to="/">
+        <div>
+          <FormattedMessage id="app.name"/>
+        </div>
+      </Link>
+      <Link to="/categories">
+        <div>
+          <FormattedMessage id="app.categories"/>
```
Comment (Terapool) [inline src/components/body/footer/Footer.tsx:20]: Pleasewrite consistently:

with space before closing of tag symbol or with it:

 id="app.categories" /&gt;
 
 or
 
  id="app.categories"/&gt;
  
  You can setup a plugin to fix such errors

File: src/components/body/header/Header.css [hunk 4]
```diff
@@ -1,8 +1,261 @@
...
-    justify-content: space-evenly;
-    align-items: center;
+.side-bar {
+  position: relative;
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  height: 64px;
+  background: #FFFFFF;
+  border: 1px solid gray; /*for exam after delete border*/
```
Comment (Terapool) [inline src/components/body/header/Header.css:8]: remove this comment - it is not easy to understand/ I do not see any reason to hold it...

File: src/components/body/header/Header.css [hunk 5]
```diff
@@ -1,8 +1,261 @@
...
+}
+
+.avatar-login {
+  display: none;
+}
+
+.button-login {
+  width: 115px;
+  height: 32px;
+  font-family: 'Apercu Pro';
```
Comment (Terapool) [inline src/components/body/header/Header.css:141]: 'Apercu Pro' is not a standard font family. If you use such families you need declare font face with source to the font

File: src/components/body/header/Header.css [hunk 6]
```diff
@@ -1,8 +1,261 @@
...
+}
+
+.user-item {
+  display: flex;
+  align-items: center;
+}
+
+.user-button {
+  width: 141px;
+  border: 1px solid #455FCE;
```
Comment (Terapool) [inline src/components/body/header/Header.css:158]: please use constant for colors if you use them often

File: src/components/body/header/Header.css [hunk 7]
```diff
@@ -1,8 +1,261 @@
...
+  border-radius: 28px;
+  font-family: 'Apercu Pro';
```
Comment (Terapool) [inline src/components/body/header/Header.css:160]: 'Apercu Pro' is not a standard font family. If you use such families you need declare font face with source to the font

File: src/components/search.scss [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+  position: absolute;
+  padding-left: 23px;
+  top: 0;
+  bottom: 0;
+  flex-wrap: wrap;
+  align-content: space-around;
+}
+
+.search-input {
+  font-family: 'Apercu Pro';
```
Comment (Terapool) [inline src/components/search.scss:19]: 'Apercu Pro' is not a standard font family. If you use such families you need declare font face with source to the font

File: src/constants.ts [hunk 9]
```diff
@@ -11,4 +11,8 @@ const USERNAME = {
   min: 3,
   max: 256,
 };
-export {BACKEND_URL, EMAIL, PASSWORD, USERNAME};
+const CODE = {
+  maxLength: 5,
+}
+const DEFAULT_COUNTRY = '375';
```
Comment (Terapool) [inline src/constants.ts:17]: DEFAULT_COUNTRY_CODE

File: src/pages/CodePage.tsx [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+
+const CodeRegex = new RegExp('[0-9a-z]{5}', 'i');
+
+const validate = async (values: FormValue) => {
+  let errors: FormErrors = {};
+  if (!CodeRegex.test(values.code)) {
+    errors.code = 'An invalid character is present in the Code. ';
+  }
+  if (values.code.length < CODE.maxLength) {
+    errors.code += 'Code should be 5 characters. ';
```
Comment (Terapool) [inline src/pages/CodePage.tsx:24]: errors.code += `Code should be ${CODE.maxLength} characters.`;

File: src/pages/CodePage.tsx [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+
+interface FormValue {
+  code: string;
+}
+
+const CodeRegex = new RegExp('[0-9a-z]{5}', 'i');
+
+const validate = async (values: FormValue) => {
+  let errors: FormErrors = {};
+  if (!CodeRegex.test(values.code)) {
```
Comment (Terapool) [inline src/pages/CodePage.tsx:20]: '12345-!!!!' will pass the chek here without error - is it ok?

File: src/pages/LoginPage.tsx [hunk 12]
```diff
@@ -32,41 +32,46 @@ const emailInvalidationRules = [
...
-          dispatch(getLogin(values))
+        validate={async (values: FormValues) => {
+          let errors: FormErrors = {};
+          if (emailInvalidationRules.some(rule => rule.test(values.email))) {
+            errors.email = 'Please enter a valid email address';
+          }
+          if (!passwordRegex.test(values.password)) {
+            errors.password = `An invalid character is present in the password. Password must be between ${PASSWORD.minLength} and ${PASSWORD.maxLength} characters; upper or lower case Latin letters (a–z, A–Z); numbers from 0 to 9; symbols ! # $ % & ' * + - / = ? ^ _ \` { | } ~`;
+          }
+          if (values.password.length >= PASSWORD.maxLength || values.password.length < PASSWORD.minLength) {
```
Comment (Terapool) [inline src/pages/LoginPage.tsx:62]: 'values.password.length >= PASSWORD.maxLength'

includes the case when 
password length = PASSWORD.maxLength

in this case you do not need error, please replace with

 (values.password.length > PASSWORD.maxLength || values.password.length < PASSWORD.minLength)

File: src/pages/PhoneNumberPage.tsx [hunk 13]
```diff
@@ -2,27 +2,39 @@ import "../components/modal/modal.scss";
...
+  const [error, setError] = useState('Phone number incorrect');
+  const [phoneNumber, setPhoneNumber] = useState(DEFAULT_COUNTRY);
+  const [isError, setIsError] = useState(false);
+  const navigate = useNavigate();
 
+  const sendPhoneNumber = () => {
+    if (!error) {
+      navigate('/users/sign_in/phone_number/code');
+    } else {
+      setIsError(!!error)
```
Comment (Terapool) [inline src/pages/PhoneNumberPage.tsx:18]: else {
      setIsError(true)
      
      as you definately know here that !!error = true

File: src/store/header/headerSlice.ts [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+const headerSlice = createSlice({
+  name: 'header',
+  initialState,
+  reducers: {
+    showDefaultPage: (state) => {
+      state.isDefaultHeader = false;
+    },
+    showStudentPage: state => {
```
Comment (Terapool) [inline src/store/header/headerSlice.ts:20]: Please, write consistantly (with braces)

 showStudentPage: (state) => {

File: src/components/Email.tsx [hunk 15]
```diff
@@ -1,30 +1,50 @@
...
+               minLength={EMAIL.minLength}
+               maxLength={EMAIL.maxLength}
+               className={"email-input" + (error ? " invalid-email-input" : "")}
+               placeholder="username@gmail.com"
+               {...field}
+               required
         />
-        {error && <span className='error-Message'>{error}</span>}
+        {error && <span className='error-message'>{error}</span>}
+        {isUser && <span className='error-message'>you entered invalid</span>}
```
Comment (NatashaMikhaltsova) [inline src/components/Email.tsx:47]: Incorrect error message, please correct it (In requirements - Incorrect Email address or password)

**Review by NatashaMikhaltsova — CHANGES_REQUESTED** [review]

File: src/pages/CodePage.tsx (hunk 11 above)
Comment (Kurilo-Pavel) [reply src/pages/CodePage.tsx:20]: I can't enter '12345-!!!!', only 5 letters and numbers together.

Comment (Terapool) [reply src/pages/CodePage.tsx:20]: Ok, resolve this comment

**Review by NatashaMikhaltsova — APPROVED** [review]

**Review by Terapool — APPROVED** [review]

**Review by OxygenO22 — APPROVED** [review]

**Review by valer-sky — APPROVED** [review]

**Pull request merged**